Add Jest for testing purposes

anttimaki opened the pull request.

Oksamies reviewed (approved):
With my limited Typescript understanding; looks good 👌

anttimaki commented on apps/nextjs/utils/urlQuery.ts:
```diff
@@ -0,0 +1,53 @@
+export const getBoolean = (val: UrlQuery): boolean | undefined => {
+  const stringVal = getString(val);
+
+  switch (stringVal?.toLowerCase()) {
```
@MythicManiac The part that writes the URLs for backend requests (the data provider) only outputs one value, so caching of backend requests shouldn't be an issue.

I've no strong feelings about supporting multiple values on frontend-level. I'd wager that if we want to support only single value, it should be either `true` (for clarity) or `1` (for brevity)?

MythicManiac replied:
I suppose it's fine as is, as long as the URL building is consistent. Choosing to not support other variants would be a nice way to enforce that at least.

If I were to choose, I'd have `true` & `false` as the supported values (`false` is needed in case the default is `true` somewhere)

anttimaki commented:
@Oksamies:

* Rebased and fixed merge conflicts in package.json
* Based on Mythic's suggestion, [only "true" and "false" are processed](https://github.com/thunderstore-io/thunderstore-ui/pull/379/commits/ed6b95433771fe8146396f75221eb8a7c0cd46b3#diff-684d9fb8afbd95cc3c3f1b43a84577b05a7b0a98f3238689e0d7349cae59000cR16-R21), other values yield `undefined`
* Updated [test cases](https://github.com/thunderstore-io/thunderstore-ui/pull/379/commits/ed6b95433771fe8146396f75221eb8a7c0cd46b3#diff-c3e5bb57bd0534fe9afe2564fb1c35ee86318b80deeff3a5137a9c49844833c1R18-R26) to match the new implementation
